### ISSUE TITLE
Duplicate definitions for multicluster links on v2.11.0

### ISSUE THREAD

**Issue by JacobValdemar** [linkerd/linkerd2#7030]
### What is the issue?
I am trying to upgrade to 2.11.0 from 2.10.2. When upgrading the multi-cluster links I run into a duplicate definition error when appying the k8s manifests for the different links because the Server and ServerAuthorization has the same name for different links. I would expect the names of the objects to be prefixed or postfixed with the cluster name to avoid name clashes?

### How can it be reproduced?
```
linkerd --kubeconfig=$kubeconfig multicluster link --cluster-name link-one > link-one.yaml

linkerd --kubeconfig=$kubeconfig multicluster link --cluster-name link-two > link-two.yaml
```

### Logs, error output, etc

Both link-one.yaml and link-two.yaml would contain:

```yaml
apiVersion: policy.linkerd.io/v1beta1
kind: Server
metadata:
  namespace: linkerd-multicluster
  name: service-mirror
  labels:
    linkerd.io/control-plane-component: linkerd-service-mirror
spec:
  podSelector:
    matchLabels:
      linkerd.io/control-plane-component: linkerd-service-mirror
  port: admin-http
  proxyProtocol: HTTP/1
---
apiVersion: policy.linkerd.io/v1beta1
kind: ServerAuthorization
metadata:
  namespace: linkerd-multicluster
  name: service-mirror
  labels:
    linkerd.io/control-plane-component: linkerd-service-mirror
spec:
  server:
    name: service-mirror
  client:
    # In order to use `linkerd mc gateways` you need viz' Prometheus instance
    # to be able to reach the service-mirror. In order to also have a separate
    # Prometheus scrape the service-mirror an additional ServerAuthorization
    # resource should be created.
    meshTLS:
      serviceAccounts:
        - name: prometheus
          namespace: linkerd-viz
```
and
```yaml
apiVersion: policy.linkerd.io/v1beta1
kind: Server
metadata:
  namespace: linkerd-multicluster
  name: service-mirror-proxy-admin
  labels:
    linkerd.io/extension: multicluster
spec:
  podSelector:
    matchLabels:
      linkerd.io/control-plane-component: linkerd-service-mirror
  port: linkerd-admin
  proxyProtocol: HTTP/1
---
apiVersion: policy.linkerd.io/v1beta1
kind: ServerAuthorization
metadata:
  namespace: linkerd-multicluster
  name: service-mirror-proxy-admin
  labels:
    linkerd.io/extension: multicluster
spec:
  server:
    name: service-mirror-proxy-admin
  client:
    # for kubelet probes
    unauthenticated: true

---

```

### Environment

- Kubernetes Version: v1.20.8
- Cluster Environment: kops
- Host OS:
- Linkerd version: 2.11.0

### Possible solution

prefix or postfix the names of the objects with the cluster (link) names 

### Additional context

There might be other manifests generated by the multicluster link that also have duplicate definitions. I haven't had time to check it all

**Comment by olix0r** [comment]
@JacobValdemar Good catch! You can workaround this by stripping out the Server/ServerAuthorization resources from subsequent links; but we'll fix this in 2.11.1 and make sure we test this scenario in our multi-cluster integration tests. Thanks for the bug report :)

**Comment by JacobValdemar** [comment]
@olix0r Thank you for the swift reply and prioritizing this issue! 🚀
